module/zfs/dmu_zfetch.c

```diff
@@ -125,6 +125,8 @@ dmu_zfetch_init(zfetch_t *zf, dnode_t *dno)
 static void
 dmu_zfetch_stream_fini(zstream_t *zs)
 {
+	zfs_refcount_destroy(&zs->zs_callers);
+	zfs_refcount_destroy(&zs->zs_refs);
 	ASSERT(!list_link_active(&zs->zs_node));
```

Comment:
I'd do the ASSERT() first.  Otherwise looks good.

Reply:
Actually yeah, I agree - moving..